### ISSUE TITLE
Is there a way to turn error reporting to a high level? 

### ISSUE THREAD

**Issue by christianleger** [ikreymer/webarchiveplayer#20]
Currently, I get many silent failures while trying to read WARC files which are successfully read by the openwayback application. 

Sometimes, the webarchiveplayer fails with no errors reported. Other times, it successfully starts up, but only gives me a couple of URLs when the WARC in question contains a few dozen resources (all response records). These are visible and functional in openwayback. 

I would like to debug this myself, but since I only have a small clue about running and debugging at the python level, I'm hoping I can at least get started by setting logging/error reporting to a maximum level. Is there such a thing? Would it be a good idea? 

I am running the latest webarchiveplayer.exe from either cmd.exe or from gitbash on Windows 8.1. 

Thanks!

**Comment by ikreymer** [comment]
Hm, latest version (1.3) should be showing an error if it can't read the WARC.

The auto-generated URL list attempts to determine what URLs are pages, not list all response records.. The heuristics can be improved as this is still experimental. That is, there are more URLs than are listed (usually), as images. etc.. are not included in this list.

I have not tested this in Windows as much as Linux/OSX, but you can build webarchiveplayer and run from command line `python setup install` then `webarchiveplayer <path/to/WARC>` and it should print errors to the console (cmd.exe in windows).

Do you have an example WARC that you could share, for any of these issues?

**Comment by christianleger** [comment]
Thanks for the reply! 

I've run the app as packaged in the latest repo here, from the command line.

I do: 

> webarchiveplayer file.warc

The console output I get is: 

2016-01-08 13:39:08,959: [DEBUG]:
2016-01-08 13:39:09,049: [DEBUG]: Adding query_html: query.html
2016-01-08 13:39:09,051: [DEBUG]: CDX Surt-Ordered? True
2016-01-08 13:39:09,111: [DEBUG]: CustomCanonilizer? True
2016-01-08 13:39:09,111: [DEBUG]: FuzzyMatcher? True
2016-01-08 13:39:09,111: [DEBUG]: Adding CDX Source: CDX File - c:\users\christ~1\appdata\local\temp\cdxbru7my.cdxj
2016-01-08 13:39:09,111: [DEBUG]: Adding search_html: pagelist_search.html
2016-01-08 13:39:09,111: [DEBUG]: Adding frame_insert_html: frame_insert.html
2016-01-08 13:39:09,111: [DEBUG]: Adding not_found_html: not_found.html
2016-01-08 13:39:09,115: [DEBUG]: Adding Path Index: c:\users\christ~1\appdata\local\temp\tmp8__jqd.txt
2016-01-08 13:39:09,142: [DEBUG]: Adding HeadInsert: head_insert.html, Banner banner.html
2016-01-08 13:39:09,142: [DEBUG]: Adding Collection:
2016-01-08 13:39:09,142: [DEBUG]: Adding CDX API Handler: -cdx
2016-01-08 13:39:09,142: [DEBUG]: Adding home_html: index.html
2016-01-08 13:39:09,142: [DEBUG]: Adding error_html: error.html
2016-01-08 13:39:09,142: [DEBUG]: Adding info_json: collinfo.json
2016-01-08 13:39:09,144: [DEBUG]: **\* pywb app inited with config from "create_wb_router"!

pywb config

# 

Deleting Temps

The result is that the app comes up in a small window and opens a browser tab, and shows a handful of URLs related to the page I recorded, however the actual main page does not show up. It does show up when I use openwayback player.

I occasionally see errors, if I feed it a completely bad WARC, but what I see are actually python errors. I don't see any output from the app itself indicating why it's not happy. 

In a little while I'll see if I can put together a WARC suitable for sharing. For now I'm trying to run pywb directly and maybe check its parsing/record reading code and maybe add some logging. 

By the way, the console I'm using on windows is the gitbash console. CMD.EXE runs with the same behavior, except that no output is shown to console at all. Powershell also shows no output, and launches the app the same.

**Comment by ikreymer** [comment]
Yes, that is the normal startup logs, so it looks like its loading correctly.

The url list generated is not an exclusive list of urls, just the pages that are detected. The current detection scheme, for example, treats most urls with `?` as not top-level pages.. The plan is to improve this.

You can also enter the url directly, eg. `http://localhost:8090/replay/<url>` where `<url>` is the url you're looking for.. That should replay it if it is in the WARC even if it is not in the page list.

Is the WARC you're using coming from Webrecorder?

I'm not sure what 'openwayback player' you're referring to, there is not an equivalent player system like this that I'm aware of in openwayback that detects which urls are pages..

As for python errors, please include them.. The 1.3 version should be detecting these, but perhaps something is being missed.

**Comment by christianleger** [comment]
Sorry for being vague. The other app I'm using to replay my WARCs is [this one](https://github.com/iipc/openwayback). It's not as self-contained and simple to launch as your app, unfortunately. My impression is that it's the same as that found over at https://webrecorder.io/. 

I will post here any further python errors I see. Also, as I'm diving into the code now, I'll point out anything I find out, or anything I do that ends up being helpful to my project. 

Thanks again!
